### PR TITLE
Add solution for Build Array from Permutation

### LINKED ISSUE #11: Build Array from Permutation
## Issue Overview
Implement a solution for the "Build Array from Permutation" LeetCode problem to practice array manipulation in Rust.

## Feature Details
Create a Rust implementation for Build Array from Permutation.

Problem summary:
Given a zero-based permutation nums (0-indexed), build an array ans of the same length where ans[i] = nums[nums[i]] for each 0 <= i < nums.length.
A zero-based permutation is an array of distinct integers from 0 to n - 1 (inclusive) where each element appears exactly once.

Constraints:
- 1 <= nums.length <= 1000
- 0 <= nums[i] < nums.length
- All integers in nums are distinct

The solution should:
- Demonstrate efficient array/vector manipulation in Rust
- Include proper error handling with Result
- Use appropriate type conversions for indices
- Include comprehensive tests

## Expected Outcome
A working Rust implementation that correctly builds the permutation array with clean, efficient code, proper error handling, and thorough test coverage.

### LINKED ISSUE #21: Kids With the Greatest Number of Candies
## Issue Overview
Implement a solution for the "Kids With the Greatest Number of Candies" LeetCode problem to practice comparative array operations in Rust.

## Feature Details
Create a Rust implementation for the problem: https://leetcode.com/problems/kids-with-the-greatest-number-of-candies/description/

Problem summary:
There are `n` kids with candies. You are given an integer array `candies`, where each `candies[i]` represents the number of candies the `i`-th kid has, and an integer `extraCandies`. For each kid, determine whether giving them `extraCandies` would enable them to have the greatest number of candies among all kids. Return a boolean array `result`.

Constraints:
- n == candies.length
- 2 <= n <= 100
- 1 <= candies[i] <= 100
- 1 <= extraCandies <= 50

The solution should:
- Use idiomatic Rust array/vector operations
- Include proper error handling
- Implement efficient comparison logic
- Feature comprehensive tests

## Expected Outcome
A well-documented Rust implementation that correctly determines which kids could have the greatest number of candies, with proper error handling and test coverage.

### PR DESCRIPTION
**Pull request by zmrdltl**
Related Issues
closes #11 

PR Description
This PR adds a solution for the LeetCode "Build Array from Permutation" problem.

The implementation includes:
- Function that builds a new array according to permutation rules: ans[i] = nums[nums[i]]
- Type safety by accepting &[usize] and returning Vec<i32>
- Safe handling of potential out-of-bounds indices using filter_map
- Marked with #[must_use] attribute to enforce using the return value

Key features:
- Comprehensive documentation with examples and usage
- Test suite covering example cases and edge cases
- Efficient implementation with iterator approach
- Integration with the existing vectors module structure

This solution maintains the code quality standards established in previous implementations while focusing on type safety and clean functional programming patterns.

### CODE REVIEW

File: src/vectors/build_array_from_permutation.rs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+/// use rust_leetcode::vectors::build_array_from_permutation;
+///
+/// let nums = vec![0, 2, 1, 5, 3, 4];
+/// assert_eq!(build_array_from_permutation(&nums), vec![0, 1, 2, 4, 5, 3]);
+/// ```
+#[must_use]
+pub fn build_array_from_permutation(nums: &[usize]) -> Vec<i32> {
+    nums.iter()
+        .filter_map(|&num| {
+            if num < nums.len() {
```
Comment (kimhanbeom) [inline src/vectors/build_array_from_permutation.rs:31]: get을 사용하면 index를 벗어나느 것에 대해서 None으로 처리해주니 if문은 제거해주세요

**Review by kimhanbeom — CHANGES_REQUESTED** [review]

**Review by kimhanbeom — APPROVED** [review]
